### ISSUE TITLE
Tensorflow 1.11 breaks simple keras estimator: global step does not increment

### ISSUE THREAD

**Issue by vijaykyr** [tensorflow/tensorflow#22850]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: running on colab.sandbox.google.com
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.11
- **Python version**: 3.6

### Describe the problem

The below code illustrates a minimal keras model that is then converted to an estimator using tf.keras.estimator.model_to_estimator() and and run using tf.estimator.train_and_evaluate().

The code runs properly up until Tensorflow 1.10, however upon upgrading to Tensorflow 1.11 the global step fails to increment, resulting in training never ending. 

You can reproduce by the running the below code on colab.sandbox.google.com, observe it doesn't increment. Then downgrade to 1.10 and observe it works as expected.

### Source code / logs

```python
import shutil
import numpy as np
import tensorflow as tf
print(tf.__version__)

from tensorflow import keras
from tensorflow.python.keras.layers import Dense

#Hyperparameters
units=64
num_classes=3
batch_size=128
model_dir='model_files'


### 1. Generate Data
data = np.random.random((1000, 100))
labels = np.random.randint(3, size=(1000, 1))

### 2. Input Function
def input_fn(features, labels, batch_size, mode):
    dataset = tf.data.Dataset.from_tensor_slices((features, labels))
      
    if mode == tf.estimator.ModeKeys.TRAIN:
      dataset = dataset.repeat()

    # Shuffle, repeat, and batch the examples.
    return dataset.batch(batch_size)

### 3. Model Code
model = keras.models.Sequential()
model.add(Dense(units=units, input_shape=(100,), activation='relu',))
model.add(Dense(units=num_classes, activation='softmax'))
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])


### 4. Estimator Code
shutil.rmtree(model_dir, ignore_errors=True) # start fresh each time
estimator = tf.keras.estimator.model_to_estimator(keras_model=model, model_dir=model_dir)


# Create TrainSpec
train_spec = tf.estimator.TrainSpec(
    input_fn=lambda: input_fn(
        data,
        labels,
        batch_size,
        mode=tf.estimator.ModeKeys.TRAIN),
    max_steps=100
)

# Create EvalSpec
eval_spec = tf.estimator.EvalSpec(
    input_fn=lambda: input_fn(
        data,
        labels,
        batch_size,
        mode=tf.estimator.ModeKeys.EVAL),
    steps=None
)

# Start training
tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
```

**Comment by PhilipXue** [comment]
The same error occurred on examples running within docker image `tensorflow:tensorflow:1.11.0-gpu-py3`

**Comment by karmel** [comment]
I thought this was fixed-- @k-w-w , is this the issue with cloning/iterator that you fixed? When did the fix go out?

**Comment by k-w-w** [comment]
This should be fixed in 1.11.0-rc1, colab runs at 1.11.0. 

@karmel how long will it take for the tensorflow version to be updated?

**Comment by karmel** [comment]
Presumably it will happen with the 1.12 release, which is coming out soon.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @karmel, @ispirmustafa: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
Closing, as this has been fixed.
